torchvision/prototype/datasets/utils/_resource.py

```diff
@@ -57,7 +54,42 @@ def _extract(file: pathlib.Path) -> None:
     def _decompress(file: pathlib.Path) -> None:
         _decompress(str(file), remove_finished=True)
 
-    def _loader(self, path: pathlib.Path) -> IterDataPipe[Tuple[str, IO]]:
+    @classmethod
+    def from_http(cls, url: str, *, file_name: Optional[str] = None, **kwargs: Any) -> OnlineResource:
+        return cls(url, file_name=file_name or pathlib.Path(urlparse(url).path).name, **kwargs)
+
+    @classmethod
+    def from_gdrive(cls, id: str, **kwargs: Any) -> OnlineResource:
+        return cls(f"https://drive.google.com/uc?export=download&id={id}", **kwargs)
```

Comment:
By using the `OnlineReader` datapipe, we no longer need different download functionality for HTTP or GDrive. Thus, I was able to remove the subclasses and replace them with these thin instantiation class methods.

If we switch to `iopath` in the future, we can probably remove them as well.